trestle/core/rules_validator.py

```diff
@@ -0,0 +1,119 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2023 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validate by confirming rule parameter values are consistent."""
+import logging
+import pathlib
+from typing import List
+
+from trestle.common.common_types import TopLevelOscalModel
+from trestle.common.common_types import TypeWithSetParams
+from trestle.common.list_utils import as_list, deep_set, get_item_from_list
+from trestle.core.catalog.catalog_interface import CatalogInterface
+from trestle.core.profile_resolver import ProfileResolver
+from trestle.core.validator import Validator
+from trestle.oscal.ssp import ByComponent, ImplementedRequirement, Statement, SystemComponent, SystemSecurityPlan
+
+logger = logging.getLogger(__name__)
+
+
+class RulesValidator(Validator):
+    """Validator to confirm all rule parameter values are consistent."""
+
+    rule_param_values_dict = {}
```

Comment:
Now I would be very careful with the class attributes. Class attributes are shared across all instances of the RulesValidator. What it means is that this dictionary will be never reset between validating different ssps. Is this intended behaviour?

Reply:
yes please don't place things at the top as attributes of the class.  Simplest is simply to define it in `model_is_valid` and pass as argument to the other methods.  You may also be able to add an `__init__` to the class and add it as a private member that the other functions refer to.

Reply:
done